src/api/firebase.js

```diff
@@ -86,12 +89,18 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	}
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listId, itemId, totalPurchases) {
+	try {
+		const itemRef = doc(db, listId, itemId);
+		await updateDoc(itemRef, {
+			dateLastPurchased: new Date(),
+			totalPurchases: totalPurchases + 1,
+		});
+	} catch (error) {
+		// Handle the error here
+		console.error('Error updating item:', error);
+		throw error; // Re-throwing the error if you want to handle it later
```

Comment:
If we're catching and re-throwing the error, then we'll likely end up with a white-screen in the event of an error from firebase. We may want to consider just logging the error for now and then re-throwing once we have error handling in place inside of `ListItem`